internal/api.go

```diff
@@ -0,0 +1,51 @@
+package internal
+
+import (
+	"net/http"
+	"github.com/go-chi/chi/v5"
+)
+
+var router *chi.Mux
+
+func createRouter() *chi.Mux{
+	if router == nil {
+		router = chi.NewRouter()
+		router.Get("/config", getConfig)
+		router.Get("/camiones", getCamiones)
+		router.Post("/camiones", createCamion)
+		router.Get("/suministros", getSuministros)
+		router.Post("/suministros", createSuministro)
```

Comment:
Tienes que tratar de tener en cuenta el concepto de recurso. Si usas el plural, es que estás pidiendo que se cree un recurso único. Pero para acceder a ese recurso, usarás el singular. Por eso, `POST` usa plural (porque quieres crear un único suministro) pero para acceder a cada recurso creado, deberás usar el singular con `GET`. ¿Cómo se accede a todos los URIs? El que se acceda o no depende de las historias de usuario. Pero para resolverlo puedes crear rutas específicas para hacer `GET` sobre todos o sobre una parte de ellos.

Reply:
Yo lo estaba pensando de la siguiente forma: Como lo que hacemos es coger todos los camiones y todos los suministros y asignarlos, pues estaba pensando en el todo, pero claro, habría pasos intermedios en el que si  necesitas consultar, por ejemplo, si tengo un suministro de X tipo, necesitaré un camión de X tipo, entonces habría que considerar también el `GET` por atributos. ¿Tendría que crear `GET /camiones/{tipoCamion}`... así con los tres atributos y en suministro y camion? Después, siguiendo las historia de usuario 1,que es la vigente, entiendo que un `GET suministros/{direccion}` no hay que hacerlo en este caso. 

La siguiente duda que me viene es, claro, el `GET` por un único atributo en nuestro problema se podría quedar corto (*en el sentido de que quieres consultar camiones del tipo X, por encima de un peso y volumen dados*), ¿Sería conveniente hacer `GET`por más de un atributo?

De todas formas me gustaría verlo el viernes en clase y, si puede ser, escuchar la explicación del 9 (mañana no voy a poder asistir porque el profesor de la siguiente hora ha decidido poner la exposición que teníamos de forma online y mi grupo es el primero, así que tendría que estar en casa a las 10.30 exactas para entrar en el meet y vivo lejos de la ETSIIT 😓)

Reply:
> Yo lo estaba pensando de la siguiente forma: Como lo que hacemos es coger todos los camiones y todos los suministros y asignarlos, pues estaba pensando en el todo, pero claro, habría pasos intermedios en el que si necesitas consultar, por ejemplo, si tengo un suministro de X tipo, necesitaré un camión de X tipo, entonces habría que considerar también el `GET` por atributos. ¿Tendría que crear `GET /camiones/{tipoCamion}`... así con los tres atributos y en suministro y camion? Después, siguiendo las historia de usuario 1,que es la vigente, entiendo que un `GET suministros/{direccion}` no hay que hacerlo en este caso.

Conviene que conozcas, y sigas, mejores prácticas [tales como esta](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/#h-use-nouns-instead-of-verbs-in-endpoint-paths). En general, una ruta REST no es una llamada a función, sino una operación (verbo) sobre un recurso (sustantivo)
> 
> La siguiente duda que me viene es, claro, el `GET` por un único atributo en nuestro problema se podría quedar corto (_en el sentido de que quieres consultar camiones del tipo X, por encima de un peso y volumen dados_), ¿Sería conveniente hacer `GET`por más de un atributo?

El diseño de un API es una actividad que tiene cierta importancia, y por eso tiene un objetivo específico. Como en casi cualquier cosa en desarrollo, hay que separar interfaz de implementación; un API REST es autónomo y podría haber infinitas formas de implementarlo por debajo. Por eso hay que, una vez más, echar mano de DDD y pensar en qué recursos se van a crear y manejar. 

> 
> De todas formas me gustaría verlo el viernes en clase y, si puede ser, escuchar la explicación del 9 (mañana no voy a poder asistir porque el profesor de la siguiente hora ha decidido poner la exposición que teníamos de forma online y mi grupo es el primero, así que tendría que estar en casa a las 10.30 exactas para entrar en el meet y vivo lejos de la ETSIIT 😓)

OK, lo vemos mañana.

Reply:
Resuelto en commit b3a6928dd23a7a417a1ace3308eba652a429d969